ote/src/cljs/ote/views/admin/monitor.cljs

```diff
@@ -81,55 +82,68 @@
+          tertiili-types (:tertiili-types monitor-data)
+          chart-type (get-in app [:admin :monitor :report-type])]
+      [:div
+       [:h1 " Valvontanäkymä "]
+       [:p "Valvontanäkymässä voit tarkastella liikkumispalveluiden tarjoajien kokonaismäärän
               kehitystä kokonaisuutena, sekä jaoteltuna liikkumispalvelutyyppien mukaan."]
-         [:p "Termi "
-          [:strong "Liikkumispalveluiden"] " tarjoajat tarkoittaa sekä NAPiin rekisteröityneitä "
-          [:strong " liikkumispalveluiden tuottajia"] ", että sellaisia yrityksiä, joista on maininta NAP:ssa
-          julkaistuiden palveluiden tiedoissa (ns. " [:strong "palveluiden tuottamiseen osallistuvat yritykset"] ").
-          Esimerkki palvelun tuottamiseen osallistuvasta yrityksestä on yksityinen elinkeinon harjoittaja,
-          joka ajaa ainoastaan taksivälityskeskuksen kyytejä."
-          ]
-         [:h2 " Liikkumispalveluiden tarjoajien lukumäärän kehitys kuukausittain "]
-         [ui-common/linkify " /admin/reports/monitor/csv/monthly-companies " " Lataa kuvaajan tiedot CSV:nä "]
-         [:div {:style {:width "75%"}}
-          [barchart-inner " bar-companies-by-month " nil companies-by-month-data]]
+       [:p "Termi "
+        [:strong "Liikkumispalveluiden tarjoajat"] " tarkoittaa sekä NAPiin rekisteröityneitä "
+        [:strong " liikkumispalveluiden tuottajia"] ", jotka vain " [:strong "osallistuvat palveluiden tuottamiseen. "]
+        "Palvelun tuottamiseen osallistuva yritys on esimerkiksi yksityinen elinkeinon harjoittaja, joka ajaa ainoastaan taksivälityskeskusken kyytejä."]
 
-         [:h2 " Liikkumispalveluiden tarjoajien lukumäärän kehitys tertiileittäin "]
-         [ui-common/linkify " /admin/reports/monitor/csv/tertiili-companies " " Lataa kuvaajan tiedot CSV:nä "]
-         [:div {:style {:width "75%"}}
-          [barchart-inner " bar-companies-by-tertiili " nil companies-by-tertiili-data]]
+       [:div {:style {:display "flex" :justify-content "flex-end" :flex-wrap "wrap"}}
+        [form-fields/field
+         {:label "Valitse raporttien tyyppi"
+          :type :selection
+          :update! #(e! (monitor-controller/->ChangeReportType %))
+          :show-option (tr-key [:admin-page :report-types])
+          :options [:tertiili :month]}
+         (get-in app [:admin :monitor :report-type])]]
 
-         [:h2 " Liikkumispalveluiden tarjoajien lukumäärä jaoteltuna liikkumispalvelutyypin mukaan kuukausittain"]
-         [:p "Yhdellä liikkumispalveluita tarjovalla palveluntuottajalla voi olla monta erilaista palvelua. Tästä syystä
-         palvelutyypeittäin jaottelu tuottaa eri lukumäärän palveluntuottajia."]
-         [ui-common/linkify " /admin/reports/monitor/csv/monthly-companies-by-service-type " " Lataa kuvaajan tiedot CSV:nä "]
-         [:div {:style {:width "75%"}}
-          [barchart-inner " bar-type-by-month " "right" monthly-types]]
+       (when (= chart-type :month)
```

Comment:
When only one case should be executed, it's it's more logical and also better performance to use `case` instead of multiple independent condition structures.
So instead of 
`when 1 a
when 2 b
when 3 c`
==>
`case
         1 a
         2 b
         3 c
         :else`

Reply:
Changed to use one if